src/ParameterRegistry.h

```diff
@@ -0,0 +1,216 @@
+
+#pragma once
+
+#include "Config.h"
+#include "Parameter.h"
+#include <map>
+#include <memory>
+#include <vector>
+
+enum ParameterSection {
```

Comment:
a parameter section seems to have an ID and a name - for me that woukld be enough to make it a `class` or at least ` struct`